dask/dataframe/dask_expr/_util.py

```diff
@@ -95,6 +95,16 @@ def is_scalar(x):
     return not isinstance(x, Expr)
 
 
+def _columns_equal(left_columns, right_columns):
+    # Checks if left_columns and right_columns are equal.
+    # It is possible that one of the arguments is a
+    # numpy array or a numpy scalar. Therefore, we
+    # cannot always rely on the == operator.
+    if is_scalar(left_columns) == is_scalar(right_columns):
```

Comment:
it looks like this disables list comparisons?

Reply:
It checks that both scalars, or both are **not** scalars. Otherwise, we don't need to compare.

I don't love this logic either, but nothing elegant jumped out at me :/

Reply:
can you just check if both are scalars and then convert? Otherwise fall back to ==

If we have numpy array in there, then this **should** raise

Reply:
>If we have numpy array in there, then this should raise

Yeah, that totally makes sense.

Reply:
>can you just check if both are scalars and then convert? Otherwise fall back to ==

We need to account for the comparison between a `np.int64(0)` and a list like `[0, 1, 2]`. The other alternatives are:

- We validate the scalar argument in `FrameBase.__getitem__` and extract the python value
- We declare that this isn't actually a bug, and that the user now needs to do `df[df.columns[0].item()]`

Reply:
I think the get item solution (and for loc) makes the most sense to me

Reply:
Okay, good point about `loc` - I moved the fix into `__getitem__` and `LocIndexer`.